althea_kernel_interface/src/open_tunnel.rs

```diff
@@ -35,11 +34,16 @@ fn is_link_local(ip: IpAddr) -> bool {
 }
 
 /// socket to string with interface id support
-fn socket_to_string(endpoint: &SocketAddr, interface_name: String) -> String {
+fn socket_to_string(endpoint: &SocketAddr, interface_name: Option<String>) -> String {
     match endpoint {
         &SocketAddr::V6(endpoint) => {
             if is_link_local(IpAddr::V6(endpoint.ip().clone())) {
-                format!("[{}%{}]:{}", endpoint.ip(), interface_name, endpoint.port())
+                format!(
+                    "[{}%{}]:{}",
+                    endpoint.ip(),
+                    interface_name.expect("Link local without interface"),
```

Comment:
Why on earth do we need to accept an `Option` as parameter if we're not willing to deal with a `None` here?

This clearly shows that the responsibility for the interface name lies on caller's part and **creates an avoidable unexpected `panic!()` opportunity for the program**

Reply:
the point of doing an unwrap in this function is that the caller should either:
1. have a ipv6 link local with a interface name
2. have a ipv6 global address with no interface name
2. have a ipv4 global address with no interface name

It would be more complex (I think) to handle this outside the function, as this assumption should be true (basically) all the time.

Reply:
the logic of the function below is we either get a valid device name from the ip (link local neighbor), or we pass in the external nic (which can either be Some or None (depending on client or gateway))

Reply:
You're right, sorry. Could you make this function return a Result though? This is a library crate after all and it would be nicer of us to leave the error for the library *user* to handle.